ray_on_golem/server/services/golem/golem.py

```diff
@@ -376,7 +387,12 @@ async def _create_activity(
         logger.info(f"Creating new activity...")
 
         await add_state_log("[1/9] Getting agreement...")
-        agreement = await stack.agreement_manager.get_agreement()
+
+        try:
+            agreement = await stack.agreement_manager.get_agreement()
+        except Exception as e:
+            logger.error(f"Creating new activity failed with `{e}`")
+            raise RuntimeError(e) from e
```

Comment:
Is `RuntimeError` the best here?

Reply:
This comment was created after the change. :D

It was changed to `GolemException` for now.